### DISCUSSION TITLE
[v3] navbar disappears on wire:navigate.hover

### DISCUSSION THREAD

**Discussion by ghost** [livewire/livewire#5966] [7. Bugs (archived)]
If there's one great thing about v3, it's wire:navigate. Period! I have tested and the features is insane, there's almost no load times 🥇 

I built a little app just to test this feature and I think I found a bug.

web.php
`Route::get('/create', fn() => view('livewire.create.index'))->name('create');`

view livewire.create.index
`@extends('layouts.app')
   @section('title', config('app.name'))
   @section('content')
        bla bla bla...
   @endsection
`

layout layouts.app
it's a bootstrap 5 layout

For example, I'm in homepage, if I hover in `<a>` to go to profile page, I can see the request being sent in inspect tools, if I click, it load in no time. So far so good! Now, if I click in an `<a>` to go again to home page , the page navbar disappears... Keep in mind that all `<a>` described has `wire:navigate.hover`.

By the way, in inspection tools I got tons of errors if I hover more than 1 time in different links.
`livewire.js?id=578b80d0:5471  Uncaught ReferenceError: clear is not defined
    at HTMLAnchorElement.handler4 (livewire.js?id=578b80d0:5471:9)
handler4 @ livewire.js?id=578b80d0:5471
livewire.js?id=578b80d0:5471  Uncaught ReferenceError: clear is not defined
    at HTMLAnchorElement.handler4 (livewire.js?id=578b80d0:5471:9)
handler4 @ livewire.js?id=578b80d0:5471
livewire.js?id=578b80d0:5471  Uncaught ReferenceError: clear is not defined
    at HTMLAnchorElement.handler4 (livewire.js?id=578b80d0:5471:9)
handler4 @ livewire.js?id=578b80d0:5471
livewire.js?id=578b80d0:5471  Uncaught ReferenceError: clear is not defined
    at HTMLAnchorElement.handler4 (livewire.js?id=578b80d0:5471:9)
handler4 @ livewire.js?id=578b80d0:5471
livewire.js?id=578b80d0:5471  Uncaught ReferenceError: clear is not defined
    at HTMLAnchorElement.handler4 (livewire.js?id=578b80d0:5471:9)
handler4 @ livewire.js?id=578b80d0:5471
livewire.js?id=578b80d0:5471  Uncaught ReferenceError: clear is not defined
    at HTMLAnchorElement.handler4 (livewire.js?id=578b80d0:5471:9)
handler4 @ livewire.js?id=578b80d0:5471
`

Example: OK
[https://imgur.com/a/rhZYNGP](url)

Example: NOT OK
[https://imgur.com/a/zJVxdwg](url)

I can provide more data if needed.

Best regards to all!
Helder

**Comment by joshhanley** [comment]
@orbitur-helderneves can you share what your component's view looks like (formatted property and not in one line)?

**Comment by ghost** [comment]
Hello @joshhanley , theres the component, if you need, I can share the template.
Best regards,
Helder
![component](https://github.com/livewire/livewire/assets/120577956/4280ad5e-78bf-4dff-9597-77358a38e127)


`
@extends('layouts.main.app')

@section('title', config('app.name'))

@section('content')

    <div class="pb-5">
        <div class="row g-4">
            <div class="col-xxl-12">
                <div class="mb-0">
                    <h2 class="mb-2">Page Title</h2>
                    <h5 class="text-700 fw-semi-bold">Page description</h5>
                </div>
            </div>
            <div class="col-xxl-12">
                <div class="card shadow-lg">
                    <div class="card-body">

                        teste page!

                    </div>
                </div>
            </div>
        </div>
    </div>

@endsection


`

**Comment by joshhanley** [comment]
@orbitur-helderneves so the issue is Livewire handles the layout for you, so you need to remove the `@extends` and `@section` from your view and make use of the `->extends()` method in the render method of your component. See this section for details (scroll down a bit) https://laravel-livewire.com/docs/2.x/rendering-components#custom-layout

Hope this helps!

**Reply by joshhanley** [reply]
@orbitur-helderneves also in future, please don't paste screenshots of code. Instead paste code in code blocks.
When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

**Comment by ghost** [comment]
Hello, first of all, thanks for the tips. I followed what you said and the problem persists. Let me show the code/tests.

web.php
```php
Route::get('/home', \App\Livewire\Home::class)->name('home');
```

livewire.php (config)
```php
'layout' => 'layouts.orbitur.app',
```

home component (livewire) - test 1
```php
class Home extends Component
{
    public function render()
    {
        return view('livewire.home')
            ->layout('layouts.orbitur.app');
    }
}
```

home component (livewire) - test 2
```php
class Home extends Component
{
    public function render()
    {
        return view('livewire.home')
            ->extends('layouts.orbitur.app')
            ->section('content');
    }
}
```

home component (livewire) - test 3
```php
class Home extends Component
{
    public function render()
    {
        return view('livewire.home')
            ->extends('layouts.orbitur.app');
    }
}
```

Layout has (to be sure lol):
```blade
        @yield('content')

        @isset($slot)
            {{ $slot }}
        @endisset
```

Layout navbar logo has an <a> to homepage:
```blade
<a wire:navigate.hover class="navbar-brand me-1 me-sm-3" href="{{ route("home") }}">
```

Analising the dev tool, I get the error:
```
livewire.js?id=777821c0:5473  Uncaught ReferenceError: clear is not defined
    at HTMLAnchorElement.handler4 (livewire.js?id=777821c0:5473:9)
handler4 @ livewire.js?id=777821c0:5473
```

**Reply by joshhanley** [reply]
@orbitur-helderneves thanks! Hmm ok can you make sure you update to beta.5 and try again, we fixed a bug, see PR #6003, with `clear` so hopefully that fixes it 🙂

**Reply by ghost** [reply]
Hello @joshhanley , I'm sorry to inform that the problem persists and the js error is still there. I'm using v3-beta5 and cache cleared.

**Reply by joshhanley** [reply]
@orbitur-helderneves thanks for testing! I'm not sure what is going on then. Any chance you could submit a failing test PR that demonstrates the issue so we can investigate? Thanks!
